src/jit/codegenxarch.cpp

```diff
@@ -5172,16 +5172,17 @@ void CodeGen::genTransferRegGCState(regNumber dst, regNumber src)
 //     pass in 'addr' for a relative call or 'base' for a indirect register call
 //     methHnd - optional, only used for pretty printing 
 //     retSize - emitter type of return for GC purposes, should be EA_BYREF, EA_GCREF, or EA_PTRSIZE(not GC)
```

Comment:
After reviewing the code, there is a GC hole and a bug in the genEmitGSCookieCheck. New Issue already opened - #3382.